plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -401,12 +402,15 @@ public DeltaLakeTableHandle getTableHandle(ConnectorSession session, SchemaTable
 
         TableSnapshot tableSnapshot = metastore.getSnapshot(tableName, session);
         Optional<MetadataEntry> metadata = metastore.getMetadata(tableSnapshot, session);
```

Comment:
you could use `.orElseThrow(() -> ...)` to throw an exception in case that the metadata entry is not present.